### DISCUSSION TITLE
`supabase db reset` deletes local storage buckets

### DISCUSSION THREAD

**Discussion by martindidiego** [supabase/supabase#12516] [Questions]
I noticed that `supabase db reset` doesn't just reset the database, but also wipes out any storage buckets that were created during local development. 

I'd like to avoid having to recreate the buckets each time I run `reset` if possible. Looking to hear how others work with storage buckets locally. 

Thanks!

**Comment by groyGetaway** [comment]
Yea this is because the storages info are in the database.
Schema `storage`. `storage.buckets` has the buckets info and `storage.objects` contains the objects in the buckets.
So you can script the creation of those in the seed.sql so that they are recreated everytime you do `supabase db reset`.

Personally I'm trying to figure out where the files are stored locally to prefill the storage for a better DX between devs.

Edit : example
```
INSERT INTO "storage".buckets
(id, "name", "owner", created_at, updated_at, public, avif_autodetection, file_size_limit, allowed_mime_types)
VALUES('images', 'images', NULL, '2023-01-08 18:53:40.835', '2023-01-08 18:53:40.835', true, false, NULL, NULL);
```

**Comment by jamsmd** [comment]
I'm having the same issue and the policies are also lost..

**Reply by eeston** [reply]
Policies can be saved in your migrations by adding `-s storage` to your migration command.

eg `supabase db diff -s storage --use-migra -f  add_storage_policies`

**Comment by internalG** [comment]
supabase db dump
